kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -401,6 +402,17 @@ class BatchJobSubmission(
         case e: Throwable => error(s"Error deleting the uploaded resource: $resource", e)
       }
     }
+
+    if (session.isExtraResourcesUploaded) {
+      // cleanup extra resources
+      extraResources.values.foreach { extraResource =>
```

Comment:
if the `extraResources` field is only used when cleaning up.

Could you delete the session upload path directly?

Reply:
+1 for this idea. We can upload all resources into `$BASE_DIR/<session_id>/` and clean the folder recursively, then we don't need to track the whole resource file list.

Reply:
Good suggestion. Please review the latest version. It puts all the uploaded files of resource and extra resources into `$BASE_DIR/<batch_id>/` folder which will be cleanup afterwards.